app/lib/actions/recurring_action.rb

```diff
@@ -17,7 +17,8 @@ def trigger_repeat(execution_plan)
       if execution_plan.delay_record && recurring_logic_task_group
         args = execution_plan.delay_record.args
         logic = recurring_logic_task_group.recurring_logic
-        logic.trigger_repeat_after(task.start_at, self.class, *args)
+        task_start_at = task.start_at < Time.zone.now ? Time.zone.now : task.start_at
```

Comment:
Just a nitpick; I don't like the ternary operator that much + the fact that we're evaluating `Time.zone.now` twice. What about this construct instead:

```ruby
[task.start_at, Time.zone.now].max
```

Reply:
This is so much more concise. Pushed this change.